src/config/index.ts

```diff
@@ -1,228 +1,174 @@
-  return cfg ? cfg.network.label : ''
+export const getChainById = (chainId: ChainId): ChainInfo => {
+  return getChains().find((chain) => chain.chainId === chainId) || emptyChainInfo
 }
 
-export const usesInfuraRPC = [ETHEREUM_NETWORK.MAINNET, ETHEREUM_NETWORK.RINKEBY, ETHEREUM_NETWORK.POLYGON].includes(
-  getNetworkId(),
-)
-
-export const getCurrentShortChainName = (): SHORT_NAME => getConfig().network.shortName
-
-export const getShortChainNameById = (networkId = getNetworkId()): string =>
-  getNetworkConfigById(networkId)?.network?.shortName || getCurrentShortChainName()
+export const getChainInfo = (): ChainInfo => {
+  return getChainById(_getChainId())
+}
 
-export const getNetworkIdByShortChainName = (shortName: string): ETHEREUM_NETWORK => {
-  if (!isValidPrefix(shortName)) return DEFAULT_NETWORK
-  return getNetworks().find((network) => network.shortName === shortName)?.id || DEFAULT_NETWORK
+export const getChainName = (): ChainName => {
+  return getChainInfo().chainName
 }
 
-export const getCurrentEnvironment = (): 'test' | 'production' | 'dev' => {
-  switch (APP_ENV) {
-    case 'test': {
-      return 'test'
-    }
-    case 'production': {
-      return 'production'
-    }
-    case 'dev':
-    default: {
-      // We need to check NODE_ENV calling jest outside of scripts
-      return NODE_ENV === 'test' ? 'test' : 'dev'
-    }
-  }
+export const getShortName = (): ShortName => {
+  return getChainInfo().shortName
 }
 
-export type NetworkSpecificConfiguration = EnvironmentSettings & {
-  network: NetworkSettings
-  disabledFeatures?: SafeFeatures
-  disabledWallets?: Wallets
+// CGW does not return `nativeCurrency.address` as it is `ZERO_ADDRESS`
+export const getNativeCurrency = (): ChainInfo['nativeCurrency'] => {
+  return getChainInfo().nativeCurrency
 }
 
-export const getConfig = (): NetworkSpecificConfiguration => {
-  const currentEnvironment = getCurrentEnvironment()
+const usesInfuraRPC = (): boolean => {
+  const { rpcUri } = getChainInfo()
+  return rpcUri.authentication === RPC_AUTHENTICATION.API_KEY_PATH
+}
 
-  // lookup the config file based on the network specified in the NETWORK variable
-  const configFile = networks[getNetworkName().toLowerCase()]
-  // defaults to 'production' as it's the only environment that is required for the network configs
-  const networkBaseConfig = configFile.environment[currentEnvironment] ?? configFile.environment.production
+export const getRpcServiceUrl = (): string => {
+  const { rpcUri } = getChainInfo()
+  const usesInfuraRPC = rpcUri.authentication === RPC_AUTHENTICATION.API_KEY_PATH
+  return usesInfuraRPC && INFURA_TOKEN ? `${rpcUri.value}${INFURA_TOKEN}` : rpcUri.value
+}
 
-  return {
-    ...networkBaseConfig,
-    network: configFile.network,
-    disabledFeatures: configFile.disabledFeatures,
-    disabledWallets: configFile.disabledWallets,
-  }
+export const getSafeAppsRpcServiceUrl = (): string => {
+  const { safeAppsRpcUri } = getChainInfo()
```

Comment:
When using from a [safe app](https://github.com/gnosis/safe-test-app) in rinkeby the `safeAppsRpcUri.value` is 'https://rinkeby.infura.io/v3/' with slash at the end so the rpcCall ends failing

![image](https://user-images.githubusercontent.com/1576776/144278123-79877929-e99f-4893-9cf7-7f784e7260a9.png)